lib/fluent/plugin/out_mongo.rb

```diff
@@ -227,35 +184,26 @@ def format_collection_name(collection_name)
       formatted
     end
 
-    def get_or_create_collection(collection_name)
-      collection_name = format_collection_name(collection_name)
-      return @clients[collection_name] if @clients[collection_name]
-
-      @db ||= get_connection
-      if @db.collection_names.include?(collection_name)
-        collection = @db.collection(collection_name)
-        unless @disable_collection_check
-          capped = collection.capped?
-          unless @collection_options[:capped] == capped # TODO: Verify capped configuration
-            new_mode = format_collection_mode(@collection_options[:capped])
-            old_mode = format_collection_mode(capped)
-            raise ConfigError, "New configuration is different from existing collection: new = #{new_mode}, old = #{old_mode}"
+    def operate(collection, records)
+      begin
+        if @replace_dot_in_key_with
+          records.map! do |r|
+            replace_key_of_hash(r, ".", @replace_dot_in_key_with)
+          end
+        end
+        if @replace_dollar_in_key_with
+          records.map! do |r|
+            replace_key_of_hash(r, /^\$/, @replace_dollar_in_key_with)
           end
         end
-      else
-        collection = @db.create_collection(collection_name, @collection_options)
-      end
-
-      @clients[collection_name] = collection
-    end
-
-    def format_collection_mode(mode)
-      mode ? 'capped' : 'normal'
-    end
 
-    def get_connection
-      db = Mongo::MongoClient.new(@host, @port, @connection_options).db(@database)
-      authenticate(db)
+        @client[collection, @collection_options].insert_many(records)
```

Comment:
> We can get n_inserted result from response, so we can log like "xxx documents are not inserted. Maybe these documents are invalid as a BSON".

Umm, we cannot get `n_inserted` from response here.
New mongo ruby driver raises exception and return no Result type value when error occurred.
ref: https://github.com/mongodb/mongo-ruby-driver/blob/0334f3a23d7c167fffd1ea83b8b156bb2333880b/lib/mongo/bulk_write/result.rb#L177